cmd/cmd_test.go

```diff
@@ -20,20 +20,107 @@
 package cmd_test
 
 import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
 	"testing"
 
 	. "gopkg.in/check.v1"
 
 	"github.com/snapcore/snapd/cmd"
+	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/testutil"
 )
 
 func Test(t *testing.T) { TestingT(t) }
 
-type cmdSuite struct{}
+type cmdSuite struct {
+	restoreExec   func()
+	execCalled    int
+	lastExecArgv0 string
+	lastExecArgv  []string
+	lastExecEnvv  []string
+}
```

Comment:
I would add tmpdir,   oldCoreDir and newCoreDir  here, setting them up in SetUpTest and setting the latter to something  $tmpdir+".../(new|old)core"), and avoid all the c.MkDir() around